### PR TITLE
Update cli and deploy version

### PR DESCRIPTION
**Pull request by jasonnrosa**
Update CLI and deploy version to be update to date. The CLI in particular was pretty out of date. Deploy action would be going from 1.3 to 1.4 (https://github.com/marketplace/actions/hubspot-cms-deploy).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "author": "",
   "license": "Apache-2.0",
   "devDependencies": {
-    "@hubspot/cms-cli": "^2.2.0",
+    "@hubspot/cli": "^3.0.9",
```
Comment (anthmatic) [inline package.json:12]: My boilerplate knowledge is very out of date but does the CLI need to be a dep?

Comment (jasonnrosa) [reply package.json:12]: Not to my knowledge. I think this particular part pre-dates me for the boilerplate project. My assumption is that it was put in to make it easier to get started if you haven't used the CLI yet but they could also just download it globally which would probably be easier 🤷🏼‍♂️. @TheWebTech would just want to double check that we don't have any boilerplate specific documentation tied to this.

Comment (TanyaScales) [reply package.json:12]: I lean towards removing it as well - but curious to hear the answer to Jason's question about it

Comment (TheWebTech) [reply package.json:12]: No documentation I'm aware of that depends on this. Kind of an odd dependency 🤷 Jason's guess as to why it was done is probably correct.

Comment (gcorne) [reply package.json:12]: It used to be used in the GitHub deploy action before @anthmatic created the reusable action.

Comment (jasonnrosa) [reply package.json:12]: Ah got it - thank you for the clarification here @gcorne! Will toss up an update to remove this entirely from the `package.json` file here.

**Comment by jasonnrosa** [comment]
Related update in GraphQL recruiting agency project (https://github.com/HubSpot/recruiting-agency-graphql-theme/pull/86)

**Review by TanyaScales — APPROVED** [review]

**Pull request merged**